commands/curation/curationaudit_test.go

```diff
@@ -410,102 +411,137 @@ func fillSyncedMap(pkgStatus []*PackageStatus) *sync.Map {
+			// Validate the results
 			if tt.requestToError == nil {
-				require.NoError(t, curationCmd.doCurateAudit(results))
+				assert.NoError(t, err)
 			} else {
-				gotError := curationCmd.doCurateAudit(results)
-				assert.Error(t, gotError)
+				assert.Error(t, err)
 				startUrl := strings.Index(tt.expectedError, "/")
 				assert.GreaterOrEqual(t, startUrl, 0)
 				errMsgExpected := tt.expectedError[:startUrl] + config.ArtifactoryUrl +
 					tt.expectedError[strings.Index(tt.expectedError, "/")+1:]
-				assert.EqualError(t, gotError, errMsgExpected)
-			}
-			defer func() {
-				if tt.cleanDependencies != nil {
-					assert.NoError(t, tt.cleanDependencies())
-				}
-			}()
-
-			// Add the mock server to the expected blocked message url
-			for key := range tt.expectedResp {
-				for index := range tt.expectedResp[key].packagesStatus {
-					tt.expectedResp[key].packagesStatus[index].BlockedPackageUrl = fmt.Sprintf("%s%s",
-						strings.TrimSuffix(config.GetArtifactoryUrl(), "/"),
-						tt.expectedResp[key].packagesStatus[index].BlockedPackageUrl)
-				}
-			}
-			// the number of packages is not deterministic for pip, as it depends on the version of the package manager.
-			if tt.tech == techutils.Pip {
-				for key := range results {
-					result := results[key]
-					result.totalNumberOfPackages = 0
-				}
-			}
-			assert.Equal(t, tt.expectedResp, results)
-			for _, requestDone := range tt.expectedRequest {
-				assert.True(t, requestDone)
-			}
-			for _, requestDone := range tt.expectedBuildRequest {
-				assert.True(t, requestDone)
+				assert.EqualError(t, err, errMsgExpected)
 			}
+			validateCurationResults(t, tt, results, config)
 		})
 	}
 }
 
+func createCurationTestEnv(t *testing.T, basePathToTests string, testCase testCase, config *config.ServerDetails) func() {
+	_, cleanUpHome := createTempHomeDirWithConfig(t, basePathToTests, testCase, config)
+	testDirPath, cleanUpTestPathDir := testUtils.CreateTestProjectEnvAndChdir(t, filepath.Join(basePathToTests, testCase.pathToProject))
+	var cleanUpChdir func()
+	if testCase.pathToTest != "" {
+		// Set the test path as the current working directory
+		cleanUpChdir = testUtils.ChangeWDWithCallback(t, filepath.Join(testDirPath, testCase.pathToTest))
+	}
+	// Run pre test exec
+	runPreTestExec(t, testDirPath, testCase)
+	return func() {
+		if cleanUpChdir != nil {
+			cleanUpChdir()
+		}
+		cleanUpTestPathDir()
+		cleanUpHome()
+	}
+}
+
+func createTempHomeDirWithConfig(t *testing.T, basePathToTests string, testCase testCase, config *config.ServerDetails) (string, func()) {
+	tempHomeDirPath, err := fileutils.CreateTempDir()
+	assert.NoError(t, err)
+	// create .jfrog dir in temp home dir
+	jfrogDir := filepath.Join(tempHomeDirPath, ".jfrog")
+	assert.NoError(t, os.MkdirAll(jfrogDir, 0777))
+	// copy .jfrog config content from test project to temp home dir
+	assert.NoError(t, biutils.CopyDir(filepath.Join(basePathToTests, testCase.getPathToTests(), ".jfrog"), jfrogDir, true, nil))
```

Comment:
does .jfrog dir exist in all testcases? it might not be the case in the future...
maybe we should check if this dir exists

Reply:
it is needed for curation.